From 7792f64f6d4fbd67ed99e63ad7e828430adb6bac Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 11 Mar 2021 16:11:46 +0000
Subject: [PATCH] overlays: Improve the i2c-rtc,i2c_csi_dsi option

The i2c_csi_dsi parameter of the i2c-rtc overlay (added for the CM4IO
board) causes the RTC devices to be probed on the I2C0 bus appearing
on GPIOs 44 and 45. However, it didn't enable the other nodes necessary
for it to work - "dtparam=i2c_vc=on" was also required.

Fix that.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts
+++ b/arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts
@@ -252,6 +252,20 @@
 		};
 	};
 
+	fragment@101 {
+		target = <&i2c0if>;
+		__dormant__ {
+			status = "okay";
+		};
+	};
+
+	fragment@102 {
+		target = <&i2c0mux>;
+		__dormant__ {
+			status = "okay";
+		};
+	};
+
 	__overrides__ {
 		abx80x = <0>,"+0";
 		ds1307 = <0>,"+1";
@@ -273,7 +287,8 @@
 		pcf85063a = <0>,"+16";
 
 		i2c0 = <&frag100>, "target:0=",<&i2c0>;
-		i2c_csi_dsi = <&frag100>, "target:0=",<&i2c_csi_dsi>;
+		i2c_csi_dsi = <&frag100>, "target:0=",<&i2c_csi_dsi>,
+			      <0>,"+101+102";
 
 		addr = <&abx80x>, "reg:0",
 		       <&ds1307>, "reg:0",
